.pre-commit-config.yaml

```diff
@@ -1,16 +1,30 @@
 ---
-# black complains loudly that fio-histo-log-pctile.py is not a python file
-# because it isn't, so ignore it.
-exclude: (agent/bench-scripts/test-bin/fio-histo-log-pctiles.py|agent/stockpile/roles|web-server)
+# `black` complains loudly that fio-histo-log-pctile.py is not a python file,
+# which it isn't, and also complains about the long forgotten demo.py file, so
+# we ignore them both.
+#
+# We must duplicate these excludes also listed in pyproject.toml because the
+# commands below are explicitly invoked on the files added or changed.  This
+# exclude prevents those checks being run.
+exclude: (agent/bench-scripts/test-bin/fio-histo-log-pctiles\.py|web-server/v0\.3/demo\.py)
 repos:
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.7.9
+  - repo: https://github.com/pycqa/flake8
+    rev: 6.0.0
     hooks:
-     - id: flake8
-       additional_dependencies: [flake8-typing-imports==1.6.0]
+      - id: flake8
+        name: flake8 (python3)
+        language_version: python3
   - repo: https://github.com/python/black.git
-    rev: 19.10b0
+    rev: 22.12.0
```

Comment:
This was not the version of Black that we used in 0.69.  This may cause some pain for anyone who tries to make a small change to that branch.  (Although, perhaps, if they don't run the pre-commit, then they won't notice.)

Likewise, we didn't use Isort in 0.69, so that's likely to cause an analogous problem.  (I don't know about Flake8....)